linux_docker_resources/Dockerfile

```
@@ -122,6 +122,16 @@ RUN apt-get update && apt-get install --no-install-recommends -y \
   uncrustify \
   yamllint
 
+# Configure proposed source, and install python3-flake8 from it
+RUN if test \( ${UBUNTU_DISTRO} = noble \); then\
+    printf "deb http://archive.ubuntu.com/ubuntu/ "$(lsb_release -cs)"-proposed restricted main multiverse universe" >> /etc/apt/sources.list.d/ubuntu-$(lsb_release -cs)-proposed.list && \
+    printf "Package: *\nPin: release a="$(lsb_release -cs)"-proposed\nPin-Priority: -10" >> /etc/apt/preferences.d/proposed-updates && \
```

Are negative values valid here? Most of the docs I have seen have `1` as the lowest value.

Do we need an additional line upping the priority of python3-flake8 to prevent it from being downgraded downstream?
```suggestion
    printf "Package: *\nPin: release a="$(lsb_release -cs)"-proposed\nPin-Priority: -10" >> /etc/apt/preferences.d/proposed-updates && \
        printf "Package: python3-flake8\nPin: release a="$(lsb_release -cs)"-proposed\nPin-Priority: 100" >> /etc/apt/preferences.d/proposed-updates && \
```

---

Yes they are see comment below. As is now, we are installing python-flake8 first from noble and then updating it to noble-proposed so it shouldn't be downgraded afterwards.

---

* It shouldn't be downgraded because the version we have is more recent. And the default for noble sources is 500, which won't replace if version is newer.

---

Thanks. I interpreted

> 
>       P < 0
>           prevents the version from being installed

As preventing the installation even upon request but that's demonstrably not the case bsaed on the job log.